### PR TITLE
[WIP] Updates to readme

### LINKED ISSUE #43: Readme mentions 4 steps but only 3 are explicitly mentioned
This is probably a minor documentation quirk
According to the README: `The detection phase is split in four steps:`
But there are only 3 explicitly mentioned steps.

### PR DESCRIPTION
**Pull request by rsarky**
This PR aims to improve the README by adding, reorganising and editing information.
This will help people interested in this project both as contributors or as users to onboard smoothly.
Fixes #43

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 PaStA - Patch Stack Analysis
 ============================
 
+About
+------------
+
+PaStA is a tool for analysing Patch Stacks or mailboxes.
```
Comment (rralf) [inline README.md]: patch stacks

File: README.md [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+In general, a patch stack (also known as patch set) is defined as a set of patches (commits) that are developed and maintained independently of the base project.
```
Comment (rralf) [inline README.md]: remove the "defines". I don't know of an "undefined set of patches" :-)

File: README.md [hunk 3]
```diff
@@ -44,8 +50,10 @@ Getting started
 
 Running PaStA
 -------------
+### Modes
+**PaStA** has 2 main modes of operation, Mailbox mode and Patch Stack mode.
```
Comment (rralf) [inline README.md]: has two modes of operation, mailbox (mbox) mode and patch stack mode.

File: README.md [hunk 4]
```diff
@@ -55,19 +63,19 @@ updates those lists.
...
 linking them into equivalence classes is split in two different commands:
 `pasta analyse` and `pasta rate`.
 Reason for the split is the comparatively long duration of the analysation
 phase. After `pasta analyse`, you might want to reuse the results of the
 analysation and run `pasta rate` for several times on the same data set.
 
-The detection phase is split in four steps:
-1. Comparing successive versions on the patch stacks
+The detection phase is split in three steps:
+1. If in Patch Stack mode, comparing successive versions on the patch stacks
```
Comment (rralf) [inline README.md]: In patch stack mode, the comparison of successive ...

**Review by rralf — CHANGES_REQUESTED** [review]
Cool thanks! Could you just address the remaining comments?

File: README.md (hunk 2 above)
Comment (rsarky) [reply README.md]: I think you misread the statement.
> A patch stack is defined as a set of patches...

Just means that I am defining what a patch stack is.

File: README.md (hunk 3 above)
Comment (rsarky) [reply README.md]: Sure

File: README.md (hunk 4 above)
Comment (rsarky) [reply README.md]: Sure

**Comment by rsarky** [comment]
This is still WIP so I will be making some other changes too along with resolving the review comments

File: README.md (hunk 1 above)
Comment (rsarky) [reply README.md]: sure

**Comment by rralf** [comment]
I added some changes and integrated the changes on next. Thanks!

**Pull request closed without merging**